### PR TITLE
fix(curriculum): broken href link ruining translations

### LINKED ISSUE #46691: Link in stern-brocot-sequence.md doesn't work
I think it needs to be fixed. It makes the string on Crowdin incredibly confusing.

File is https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/stern-brocot-sequence.md

/cc @Sboonny

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46691

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/stern-brocot-sequence.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ dashedName: stern-brocot-sequence
 
 # --description--
 
-For this task, the Stern-Brocot sequence is to be generated by an algorithm similar to that employed in generating the <a href="<https://rosettacode.org/wiki/Fibonacci sequence>" target="_blank" rel="noopener noreferrer nofollow">Fibonacci sequence</a>.
+For this task, the Stern-Brocot sequence is to be generated by an algorithm similar to that employed in generating the <a href="https://rosettacode.org/wiki/Fibonacci" target="_blank" rel="noopener noreferrer nofollow">Fibonacci sequence</a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/stern-brocot-sequence.md:11]: this page doesn't exist

```suggestion
For this task, the Stern-Brocot sequence is to be generated by an algorithm similar to that employed in generating the <a href="https://rosettacode.org/wiki/Fibonacci_sequence" target="_blank" rel="noopener noreferrer nofollow">Fibonacci sequence</a>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-pythagorean-means.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ dashedName: averagespythagorean-means
 
 # --description--
 
-Compute all three of the <a href="<https://en.wikipedia.org/wiki/Pythagorean means> "wp: Pythagorean means"" target="_blank" rel="noopener noreferrer nofollow">Pythagorean means</a> of the set of integers $1$ through $10$ (inclusive).
+Compute all three of the <a href="https://en.wikipedia.org/wiki/Pythagorean" target="_blank" rel="noopener noreferrer nofollow">Pythagorean means</a> of the set of integers $1$ through $10$ (inclusive).
```
Comment (ieahleen) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-pythagorean-means.md:11]:
```suggestion
Compute all three of the <a href="https://en.wikipedia.org/wiki/Pythagorean_means" target="_blank" rel="noopener noreferrer nofollow">Pythagorean means</a> of the set of integers $1$ through $10$ (inclusive).
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/factors-of-a-mersenne-number.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ Further properties of Mersenne numbers allow us to refine the process even more.
 
 Any factor `q` of <code>2<sup>P</sup>-1</code> must be of the form `2kP+1`, `k` being a positive integer or zero. Furthermore, `q` must be `1` or `7 mod 8`.
 
-Finally any potential factor `q` must be <a href="<https://rosettacode.org/wiki/Primality by Trial Division> "Primality by Trial Division"" target="_blank" rel="noopener noreferrer nofollow">prime</a>.
+Finally any potential factor `q` must be <a href="https://rosettacode.org/wiki/Primality" target="_blank" rel="noopener noreferrer nofollow">prime</a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/factors-of-a-mersenne-number.md:51]:
```suggestion
Finally any potential factor `q` must be <a href="https://rosettacode.org/wiki/Primality_by_trial_division" target="_blank" rel="noopener noreferrer nofollow">prime</a>.
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

**Review by majestic-owl448 — DISMISSED** [review]

**Review by majestic-owl448 — APPROVED** [review]

**Review by jeremylt — APPROVED** [review]
All the new links work for me

**Pull request merged**